astroquery/jplhorizons/core.py

```diff
@@ -1132,6 +1141,30 @@ def vectors_async(self, *, get_query_payload=False,
         return response
 
     # ---------------------------------- parser functions
+    @staticmethod
+    def _prep_loc_dict(loc_dict, attr_name):
+        """prepare coord specification dict for 'location' or 'id'"""
+        if {'lat', 'lon', 'elevation'} - set(loc_dict.keys()) != set():
```

Comment:
An empty set is falsy and a non-empty set is truthy. And the explicit conversion of dict keys to `set` is not needed (at least in Python 3.10):
```suggestion
        if {'lat', 'lon', 'elevation'} - loc_dict.keys():
```

Reply:
Huh, I didn't know that about `dict_keys`, neat -- set operators in general fail with non-set iterables and I'd never tried. 

I have an allergy to truthiness tests but I understand that people often prefer them. Is your preference here strong?

Reply:
Actually, my preference isn't _that_ strong. I'll just change this.